### PR TITLE
update grid.param file doxygen string

### PR DESCRIPTION
**Pull request by psychocoderHPC**
Update and add in all `grid.param` files the latest doxygen string for
the file description.

Co-authored-by: @sbastrakov 

Note: I have not added the `input-output` label because this is not a change a user must know about (no breaking change).

### CODE REVIEW

**Review by steindev — APPROVED** [review]

File: share/picongpu/benchmarks/SPEC/include/picongpu/param/grid.param [hunk 1]
```diff
@@ -17,6 +17,28 @@
  * If not, see <http://www.gnu.org/licenses/>.
  */
 
+/** @file
+ *
+ * Definition of cell sizes and time step. Our cells are defining a regular,
+ * cartesian grid. Our explicit FDTD field solvers define an upper bound for
```
Comment (n01r) [inline share/picongpu/benchmarks/SPEC/include/picongpu/param/grid.param:23]: define --> _require_?

File: share/picongpu/benchmarks/SPEC/include/picongpu/param/grid.param [hunk 2]
```diff
@@ -17,6 +17,28 @@
...
+ *
+ * **Units in reduced dimensions**
+ *
+ * In 2D3V simulations, the CELL_DEPTH_SI (Z) cell length
+ * is still used for normalization of densities, etc..
+ *
+ * A 2D3V simulation in a cartesian PIC simulation such as
+ * ours only changes the degrees of freedom in motion for
+ * (macro) particles and all (field) information in z
+ * travels instantaneous, making the 2D3V simulation
```
Comment (n01r) [inline share/picongpu/benchmarks/SPEC/include/picongpu/param/grid.param:37]: _instantaneously_

**Review by n01r — COMMENTED** [review]
Just a minor wording suggestion and a grammar improvement

File: share/picongpu/benchmarks/SPEC/include/picongpu/param/grid.param (hunk 2 above)
Comment (sbastrakov) [reply share/picongpu/benchmarks/SPEC/include/picongpu/param/grid.param:37]: It's also perhaps not correct, or at least confusing. Nothing changes along z, so there is no information to travel there.

**Comment by psychocoderHPC** [comment]
ready to merge

**Review by sbastrakov — APPROVED** [review]

**Pull request merged**